section1/reflection.md

```diff
@@ -2,18 +2,37 @@
 
+Answer: When I worked at Exygy, a consultancy in San Francisco, I was taught a toxic habit of avoiding asking questions of my engineering team lead. In fact, "how often I asked for help" was used as a criteria in my performance review, and if I asked for help more often than they thought I should be, I was given a worse score. Just another one of the reasons that environment was not a good fit. In class, I am often comfortable asking questions, as long as I feel as if there's enough time for everyone. If the instructor doesn't have time to answer everyone's questions or is rushed, what choice do I have but to hold my tongue? Otherwise, I enjoy asking follow-up questions until I fully comprehend a topic. I usually also re-state the final conclusion of what I learned to the person I was asking, just to make sure we agree that I understood what they meant.
+
 2. What are the data types you learned about in this section? In your own words, define each.
+String, number, boolean
+A string holds text, including letters numbers and symbols, including special characters. A number holds numbers, including negative numbers and floating point numbers, and does not use punctuation in large numbers. A boolean holds either true or false.
+
 
 3. How would you log the string `"Hello World!"` to the console?
+console.log("Hello World!")
 
 4. What is/are the character(s) you would use to indicate comments in a JavaScript file? What is the purpose of a code comment?
+//
+/* */
+A code comment is not executed when the file is run. It can be used for elucidation in natural language, or for temporarily deactivating a portion of code
 
 5. In your own words, what is a variable? How would you explain it to a 5 year old?
 
+A variable is a container for data with a name, that acts as a label, and a value that can be assigned and changed later.
+
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
 
+screenName, password, and followerCount might be used in an app like Instagram (though I doubt they use those exact names)
+screenName would hold a string, so would password, and followerCount would hold a number
+
 7. In your own words, explain what concatenation is.
 
+Concatenation is the joining together of two things. In our case, it usually means joining one string to another using the + operator.
+
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
 
+When displaying a welcome message like "Welcome back, David!", I bet the developers used concatenation (or interpolation) to append a dynamic variable containing the user's name to the static string containing the welcome message.
+
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+Whew, okay, gotta have a question...Alright, how about this. Is a variable considered a data type?
```

Comment:
A variable is a way to store a data type, you can look into how to set data types in Java where you need to determine the data type before setting a variable. JavaScript and Ruby are a lot more fluid in this area.